.github/workflows/node.yml

```diff
@@ -1,28 +1,21 @@
 name: Node applications
 
 on:
-  push:
-    branches:
-    - master
-    paths:
-    - 'graphql/**'
-    - 'front/**'
-    - 'export/**'
-    - '.github/workflows/node.yml'
-  pull_request:
-    paths:
-    - 'graphql/**'
-    - 'front/**'
-    - 'export/**'
-    - '.github/workflows/node.yml'
+  workflow_call:
+    inputs:
+      app-dir:
+        type: string
+        required: true
 
 jobs:
   build:
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: true
-      matrix:
-        app: [front, graphql, export]
+
+    services:
```

Comment:
est-ce qu'il est possible de ne pas lancer le service MongoDB si on build/test le frontend ?

Reply:
D'ailleurs on a déjà une base de données en mémoire pour les tests https://github.com/EcrituresNumeriques/stylo/blob/master/graphql/jest/in-memory-mongodb-environment.js mais peut être que l'idée du service est plus de tester la migration des données ? Dans ce cas il faudrait qu'on lance le service graphql/backend non ?

Reply:
Y'a pas de conditionnalité au service malheureusement.

J'avais testé d'extraire la clé `services` de `node.yml` mais on ne peut pas _et_ utiliser `uses` (pour invoquer un workflow externe) _et_ invoquer un `services`.

Il faut que le service existe _soit_ dans le workflow externe, _soit_ qu'on se passe du workflow externe.

Initialiser le conteneur sans s'en servir ajoute environ 6—9 secondes sur un build qui en prend 1'10 (front), 0'26 (export) ou 0'43 (graphql).

On peut effectivement rebasculer sur le serveur in-memory (vu que c'est un véritable serveur), et lancer les migrations dedans.

Ça serait ta piste préférée ?

Reply:
Oui si ça fonctionne je pense que c'est bien. J'ai eu une bonne experience avec `mongodb-memory-server`.
Pour la version, il est apparemment possible d'instancier `MongoMemoryServer` afin de préciser la version: https://github.com/typegoose/mongodb-memory-server#available-options-for-mongomemoryserver

Reply:
Super, faisons comme ça ; ça simplifiera la config du CI et celle des tests en local également.